Remove non-kotlin files and README text.

zach-klippenstein opened the pull request:
Resolves https://github.com/square/workflow/issues/1187.

Corresponds to https://github.com/square/workflow/pull/1198.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/square/workflow-kotlin?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/square/workflow-kotlin?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/square/workflow-kotlin?pullRequest=1) it.</sub>

zach-klippenstein commented:
Not sure why the gradle verification is failing, the JAR is clearly present.

zach-klippenstein commented:
- Fixed the github workflow to not specify path filter and not try to run mkdocs.
- Fixed `lint_docs.sh` to use the correct paths and filters.
- Fixed lint issues with the now-linted README in the terminal sample.
